### PR TITLE
feat: Implement Doctor versions checking models

### LINKED ISSUE #1570: [Metrics CLI] Implement models for Doctor versions checking
**Description**
Implement `DoctorTargetValidationResult` and `DoctorValidationConclusion` models.
**Acceptance Criteria**
- [ ] `DoctorTargetValidationResult` implemented
- [ ] `DoctorValidationConclusion` implemented
- [ ] All models are properly tested

**Additional context**
[Doctor output improvement](https://github.com/Flank/flank-dashboard/blob/master/metrics/cli/docs/features/doctor_output_improvements/01_doctor_output_improvements.md)

### PR DESCRIPTION
**Pull request by ghost**
Fixes #1570 

## Checklist

- [x] Documented
- [x] Unit tested

### CODE REVIEW

File: metrics/cli/lib/cli/doctor/strings/doctor_strings.dart [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  static const String version = 'version';
+  static const String processOutput = 'Process output...';
+  static const String commandNotFound = 'command not found:';
+  static const String recommendations = 'Recommendations...';
+  static const String warning = 'Warning message...';
+  static const String commandError = 'Command Error Message...';
+  static const String versionMismatch = 'Versions mismatch does not guarantee '
+      'the successful deployment! We advise to use the recommended version'
+      ' of the tool.';
+  static String installMessage(String name, String installUrl) {
```
Comment (illia-romanenko) [inline metrics/cli/lib/cli/doctor/strings/doctor_strings.dart:19]:
```suggestion
      ' of the tool.';

  static String installMessage(String name, String installUrl) {
```

**Review by illia-romanenko — CHANGES_REQUESTED** [review]
Well done! I have a small formatting suggestion, but other than that looks good @solid-dmytrotkachuk - could you please take a look and fix it?

**Comment by solid-dmytrotkachuk** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by illia-romanenko — APPROVED** [review]
LGTM -  thanks @solid-tetianamovlian @solid-dmytrotkachuk !

**Pull request merged**